meta-client/src/test/java/org/apache/helix/metaclient/impl/zk/ZkMetaClientTestBase.java

```diff
@@ -0,0 +1,95 @@
+package org.apache.helix.metaclient.impl.zk;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.helix.metaclient.impl.zk.factory.ZkMetaClientConfig;
+import org.apache.helix.zookeeper.zkclient.IDefaultNameSpace;
+import org.apache.helix.zookeeper.zkclient.ZkServer;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+
+
+public abstract class ZkMetaClientTestBase {
+
+  protected static final String ZK_ADDR = "localhost:2183";
+  protected static final int DEFAULT_TIMEOUT_MS = 1000;
+  protected static final String ENTRY_STRING_VALUE = "test-value";
+
+  private final Object _syncObject = new Object();
+
+  private ZkServer _zkServer;
```

Comment:
An open discussion here. So with BeforeSuite, testng will only trigger that method once. That means for most subclasses of this, this `_zkServer` will be null. Maybe we should set this as static? But I'm really not sure if this is the right approach.
(And I don't know what will happen with afterSuite, could there be NPE?)